src/main/java/net/minecraftforge/event/entity/player/ItemFishedEvent.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Minecraft Forge
+ * Copyright (c) 2016.
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation version 2.1
+ * of the License.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+
+package net.minecraftforge.event.entity.player;
+
+import com.google.common.base.Preconditions;
+import net.minecraft.entity.projectile.EntityFishHook;
+import net.minecraft.item.ItemStack;
+import net.minecraft.util.NonNullList;
+
+import javax.annotation.Nonnegative;
+import java.util.List;
+
+/**
+ * This event is called when a player fishes an item.
+ * You can change the items the player will get using {@link #getItemStacks()}
+ * You can also change the damage the rod will take using {@link #setRodDamage(int)}
+ * You can use {@link #getHookEntity()} to get stuff based on the entity, like {@link EntityFishHook#fishApproachAngle}
```

Comment:
These descriptions of methods shouldn't be here. If functions are added or removed, surely updating this will be forgotten. The class documentation should just contain documentation on the class as a whole, not the specifics/implementation details—that should be left to the member documentations.